### ISSUE TITLE
Standard stdlib package list is way too broad, need to exclude (at least) go/ (and some of runtime/ too) (eg go/version and runtime/metrics)

### ISSUE THREAD

**Issue by ldemailly** [mgechev/revive#1637]
Currently it generates way too many false positives for packages that are _not_ conflicting with stdlib as despite not being internal they... are internal to the go command line or runtime metrics

_Originally posted by @ldemailly in https://github.com/mgechev/revive/issues/1593#issuecomment-3898388240_

**Comment by ldemailly** [comment]
`runtime/pprof` is fine to complain about for instance, so it's less clear... but in general going from a list of 10 before to 171 now seems bound to create havoc

**Comment by bconway** [comment]
Strong agree. I would argue that no one thinks of the Go standard library when encountering packages `user` or `token`.

**Comment by ldemailly** [comment]
got a workaround in https://github.com/mgechev/revive/issues/1602#issuecomment-3898380407 but it's not quite working yet (or I'm holding it wrong which is very much possible but my PR for config update breaks)

**Comment by ldemailly** [comment]
fixed thanks to ludovic https://github.com/fortio/workflows/pull/195/changes/6ebef1720b70ee1cd18f7e106eb18f808d35eafd

**Comment by ryancurrah** [comment]
Starting to see lots of these pop up in the latest golangci-lint release.

```
internal/ci/build/build.go:1:9: var-naming: avoid package names that conflict with Go standard library package names (revive)
internal/cmd/version/version.go:1:9: var-naming: avoid package names that conflict with Go standard library package names (revive)
internal/version/version.go:1:9: var-naming: avoid package names that conflict with Go standard library package names (revive)
```

**Comment by ldemailly** [comment]
I think reverting to a manually curated list might be best ?

**Comment by ryancurrah** [comment]
These seem to be valid std libs. I think we are just going to turn off this lint rule though since we have 1000s of pre-existing packages that now trigger this lint rule it's not feasible to go to back and change the package names. Also when Go introduces new stdlib packages now we have to rename ours, it just doesnt make sense.

**Comment by ryancurrah** [comment]
Heres an example revive config we created using a manually curated list as @ldemailly suggested.

```
    revive:
      enable-default-rules: true
      rules:
        - name: var-naming
          arguments:
            - [] # AllowList
            - # DenyList - Manually curated list of stdlib packages to avoid naming conflicts
              - archive
              - bufio
              - bytes
              - cmp
              - compress
              - container
              - context
              - crypto
              - database
              - encoding
              - errors
              - expvar
              - flag
              - fmt
              - hash
              - html
              - http
              - image
              - index
              - io
              - iter
              - json
              - jsontext  # New (Go 1.25)
              - log
              - maps
              - math
              - mime
              - net
              - os
              - path
              - pbkdf2    # New (Go 1.24)
              - plugin
              - reflect
              - regexp
              - runtime
              - secret    # New (Go 1.26 - high collision risk)
              - sha3      # New (Go 1.24)
              - slices
              - sort
              - strconv
              - strings
              - sync
              - synctest  # New (Go 1.25)
              - syscall
              - template
              - testing
              - text
              - time
              - unicode
              - unsafe
              - url
              - weak      # New (Go 1.24)
            - - skip-package-name-collision-with-go-std: true
```
